server/service/osquery_utils/queries.go

```diff
@@ -1114,7 +1115,30 @@ func directIngestMDMMac(ctx context.Context, logger log.Logger, host *fleet.Host
 		}
 	}
 
-	return ds.SetOrUpdateMDMData(ctx, host.ID, false, enrolled, rows[0]["server_url"], installedFromDep, "")
+	return ds.SetOrUpdateMDMData(ctx,
+		host.ID,
+		false,
+		enrolled,
+		rows[0]["server_url"],
+		installedFromDep,
+		deduceMDMNameMacOS(rows[0]),
+	)
+}
+
+func deduceMDMNameMacOS(row map[string]string) string {
```

Comment:
I wonder if this and `deduceMDMNameWindows` should live alongside `fleet.MDMNameFromServerURL` or  vise-versa.